events/event_nettrace.go

```diff
@@ -16,6 +20,24 @@ var NetTraceIntegrator = func() func(basictracer.SpanEvent) {
 			tr.SetMaxEvents(1000)
 		case basictracer.EventFinish:
 			tr.Finish()
+		case basictracer.EventTag:
+			tr.LazyPrintf("%s:%v", t.Key, t.Value)
+		case basictracer.EventLogFields:
+			if len(t.Fields) == 1 {
+				// Fast path when there is exactly one field.
+				tr.LazyPrintf("%s:%v", t.Fields[0].Key(), t.Fields[0].Value())
+			} else {
+				var buf bytes.Buffer
+				for i, f := range t.Fields {
+					if i > 0 {
+						buf.WriteByte(' ')
+					}
+
+					fmt.Fprintf(&buf, "%s:%v", f.Key(), f.Value())
+				}
+
+				tr.LazyPrintf("%s", buf.String())
```

Comment:
These two branches behave differently with regards to lazily-vs-eagerly evaluating field values. If we think lazy evaluation is OK in the first place, I'd like to see lazy evaluation in the second case, so something like:

type LazyTraceEmitter basictracer.EventLogFields

func (l LazyTraceEmitter) String() string {
  ...
}

...
tr.LazyLog(LazyTraceEmitter(t), false)
...
(should work for both cases)